test/IntegrationTests/Helpers/MockMetricsCollector.cs

```diff
@@ -146,7 +146,7 @@ public void AssertExpectations(TimeSpan? timeout = null)
     internal void AssertEmpty(TimeSpan? timeout = null)
     {
         timeout ??= Timeout.NoExpectation;
-        if (_metricsSnapshots.TryTake(out var metricsResource, timeout.Value))
+        if (_metricsSnapshots.TryTake(out var metricsResource, timeout.Value) && metricsResource.Count > 0)
         {
             Assert.Fail($"Expected nothing, but got: {metricsResource}");
         }
```

Comment:
We should check if all `_metricsSnapshots` are empty

I think it should be:

```suggestion
        while (_metricsSnapshots.TryTake(out var metricsResource, timeout.Value))
        {
            if (metricsResource.Count > 0)
            {
                Assert.Fail($"Expected nothing, but got: {metricsResource}");
            }
        }
```

Current code may exit too early

Reply:
@pellared the second condition will only executed if first is true...both (suggestion and what I have written) is same.. am I missing something ?

Reply:
my suggestion has `while` in the first condition to make sure to read all `_metricsSnapshots` and not just the first one

Reply:
See https://github.com/open-telemetry/opentelemetry-dotnet-instrumentation/pull/1693#discussion_r1036221654